packages/react/src/runtimes/composer/BaseComposerRuntimeCore.tsx

```
@@ -123,4 +136,27 @@ export abstract class BaseComposerRuntimeCore implements ComposerRuntimeCore {
+  private _eventSubscribers = new Map<string, Set<() => void>>();
+
+  public unstable_on(event: ComposerRuntimeEventType, callback: () => void) {
+    const subscribers = this._eventSubscribers.get(event);
+    if (!subscribers) {
+      this._eventSubscribers.set(event, new Set([callback]));
+    } else {
+      subscribers.add(callback);
+    }
+
+    return () => {
+      const subscribers = this._eventSubscribers.get(event)!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)